pdb2sql/align.py

```diff
@@ -0,0 +1,205 @@
+import numpy as np
+from .pdb2sqlcore import pdb2sql
+from .interface import interface
+from .transform import rot_xyz_around_axis
+
+
+def align(pdb, axis=None, export=True, **kwargs):
+    """Align the max principal component of a structure along one of the cartesian axis
+
+    Arguments:
+        pdb {str, pdb2sql} -- the pdbfile or the sql database of the complex
+
+    Keyword Arguments:
+        axis {str} -- cartesian axis for alignement (default: {'z'})
+        export {bool} -- export the aligned structure to file
+        **kwargs {dict} -- option to select subpart of the structure for alignement
+
+    Returns:
+        pd2sql -- sql databse of the aligned structure
+
+    Example:
+        >>> pdb = '1AK4'
+        >>> sql = align(pdb,chainID='A')
+    """
+
+    if not isinstance(pdb, pdb2sql):
+        sql = pdb2sql(pdb)
+    else:
+        sql = pdb
+
+    # extract coordinate
+    xyz = np.array(sql.get('x,y,z', **kwargs))
+
+    # get the pca eigenvect we want to align
+    vect = get_max_pca_vect(xyz)
+
+    # align the sql
+    sql = align_pca_vect(sql, vect, axis)
+
+    # export the pdbfile
+    if export:
+        export_aligned(sql)
+
+    return sql
+
+
+def align_interface(ppi, plane='xy', export=True, **kwargs):
+    """align the interface of a complex in a given plane
+
+    Arguments:
+        ppi {interface} -- sql interface or pdb file
+        plane {str} -- plane for alignement
+
+    Keyword Arguments:
+        export {bool} -- write a pdb file (default: {True})
+        kwargs {dict} -- keywaord argument from interface.get_contact_atoms method
+    """
+
+    if not isinstance(ppi, interface):
+        sql = interface(ppi)
+    else:
+        sql = ppi
+
+    index_contact = sql.get_contact_atoms(**kwargs)
+    row_id = []
+    for _, v in index_contact.items():
+        row_id += v
+    xyz = np.array(sql.get('x,y,z', rowID=row_id))
+
+    # get the pca eigenvect we want to align
+    vect = get_min_pca_vect(xyz)
+
+    # align the sql database
+    dict_plane = {'xy': 'z', 'xz': 'y', 'yz': 'x'}
```

Comment:
I feel confused here. Let's treat the interface as a cuboid with l=3, w=2 and h=1. What we want to do is align `l` to `x` and `w` to `y` for aligning interface to plane `xy`, right?

Reply:
is it ok now that we clarified that online ?

Reply:
Thanks, my fault, it's clear now.